sdk/iot/hub/src/az_iot_hub_client.c

```diff
@@ -11,8 +11,8 @@
 #include <_az_cfg.h>
 
 static const uint8_t hub_client_forward_slash = '/';
-static const uint8_t hub_client_param_separator = '&';
-static const uint8_t hub_client_param_equals = '=';
+static const az_span hub_client_param_separator_span = AZ_SPAN_LITERAL_FROM_STR("&");
+static const az_span hub_client_param_equals_span = AZ_SPAN_LITERAL_FROM_STR("=");
```

Comment:
I am not a big fan of this change which forces us to us `az_span_ptr` in more places to get back the single byte. I would have preferred this remained a `uint8_t `and we created a helper for finding a single byte.

Reply:
?? obviously we don't want two identical defines in the source code. I don't know what you means by a helper. That sounds worse.